internal/vercheck/vercheck.go

```diff
@@ -62,14 +71,84 @@ func SelfUpdate(stdOut, stdErr io.Writer) error {
 		return errors.WithStack(err)
 	}
 
-	fmt.Fprint(stdErr, "Latest version: ")
+	fmt.Fprint(stdOut, "Latest version: ")
+	return triggerUpdateAndPrintNewVersion(stdOut, stdErr)
+}
+
+// selfUpdateBinary will update the binary to the latest version.
+func selfUpdateBinary(stdOut, stdErr io.Writer) error {
+	resp, err := http.Get(fmt.Sprintf("https://releases.jetpack.io/devbox/%s/version", currentChannel))
+	if err != nil {
+		fmt.Fprint(
+			stdErr,
+			"Failed to get latest version. "+
+				"Please try again, or run the command from https://www.jetpack.io/devbox/docs/installing_devbox/",
+		)
+		return errors.WithStack(err)
+	}
+	defer resp.Body.Close()
+
+	newVersion, err := io.ReadAll(resp.Body)
+	if err != nil {
+		return errors.WithStack(err)
+	}
+
+	// Remove the version cache file, so that when we re-create it the expiration
+	// time is reset.
+	_ = os.Remove(versionCacheFilePath())
+
+	// Write the new version to the version cache file.
+	if err := os.WriteFile(versionCacheFilePath(), newVersion, 0644); err != nil {
+		return errors.WithStack(err)
+	}
+
+	return triggerUpdateAndPrintNewVersion(stdOut, stdErr)
+}
```

Comment:
This can be simplified to:

```golang
func selfUpdateBinary(stdOut, stdErr io.Writer) error {
  os.Remove(filepath.Join(xdg.CacheSubpath("devbox"), "current-version"))
  return triggerUpdateAndPrintNewVersion(stdOut, stdErr)
}
```

By removing the file and calling the launcher again you trigger a new fetch (we always fetch new version if current-version is missing even if auto update is turned off). 

A small change I can add to the launcher is to pass this file path as an ENV var (e.g. `DEVBOX_CURRENT_VERSION_FILE_PATH`) to reduce the coupling. 

A minor UX improvement could be to have `triggerUpdateAndPrintNewVersion` return the new version.

```golang
func selfUpdateBinary(stdOut, stdErr io.Writer) error {
  os.Remove(filepath.Join(xdg.CacheSubpath("devbox"), "current-version"))
  version, err := triggerUpdateAndPrintNewVersion(stdOut, stdErr)
  if err != nil {
    return err
  } 
  if semver.Compare(version, build.Version) == 0 {
    print you are already at latest version, blah...
  }
}
```